jdaviz/configs/mosviz/helper.py

```diff
@@ -258,9 +260,27 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
             msg = SnackbarMessage(msg, color='warning', sender=self)
             self.app.hub.broadcast(msg)
 
+        self.link_table_data(None)
+
+        # Any subsequently added data will automatically be linked
+        # with data already loaded in the app
+        self.app.auto_link = True
+
         # Load the first object into the viewers automatically
         self.app.get_viewer("table-viewer").figure_widget.highlighted = 0
 
+    def link_table_data(self, data_obj):
+        """
+        Batch link data in the Mosviz table rather than doing it on
+        data load.
+
+        Parameters
+        ----------
+        data_obj : obj
+            Input for Mosviz data parsers.
+        """
+        super().load_data(data_obj, parser_reference="mosviz-link-data")
```

Comment:
Just out of interest, what is the motivation for doing the linking as a parser as opposed to just a method?

Reply:
I think it was to keep all related code in one file and have the linking method be callable from the helper file, which (if I remember correctly) can only be done using `parser_reference`.